django-backend/fecfiler/reports/form_3x/views.py

```diff
@@ -43,3 +44,16 @@ def update(self, request, pk=None):
 
     def partial_update(self, request, pk=None):
         return super(ModelViewSet, self).partial_update(request, pk)
+
+    @action(
+        detail=True,
+        methods=["post"],
+        url_name="amend",
+    )
+    def amend(self, request, pk):
+        report = self.get_object()
+        report.form_type = "F3XA"
+        report.report_version = int(report.report_version or "0") + 1
+        report.upload_submission = None
+        report.save()
+        return Response(f"amended {report}")
```

Comment:
@Elaine-Krauss-TCG There's a lot of duplicate code where with the form_24 version. What if we had a utility function in reports.views.amend that F24 and F3X called from their views? It can pass the form_type as a parameter to the function.

Reply:
or you can avoid the routing issue in the app, have them all route to reports/amend and have the report instance define a function to come up with amendment info:

reports/views.py:
```
 @action(
        detail=True,
        methods=["post"],
        url_name="amend",
    )
    def amend(self, request, pk):
        report = self.get_object()
        report.amend()
        return Response(f"amended {report}")
```

reports/models.py
```
    def amend(self):
        self.form_type = self.get_form_name() + "A"
        self.report_version = int(self.report_version or "0") + 1
        self.upload_submission = None
        self.save()
```